compiler/src/iree/compiler/Codegen/Common/TileDispatchUsingInterface.cpp

```diff
@@ -390,45 +478,193 @@ FailureOr<TilingResult> TileDispatchUsingSCFForOp::returningMatchAndRewrite(
+    worklist.pop_front();
+    for (OpOperand &operand : currOp->getOpOperands()) {
+      auto tilingInterfaceProducer =
+          operand.get().getDefiningOp<TilingInterface>();
+      if (!tilingInterfaceProducer ||
+          producers.count(tilingInterfaceProducer)) {
+        continue;
+      }
+      worklist.push_back(tilingInterfaceProducer);
+      producers.insert(tilingInterfaceProducer);
     }
   }
-  for (auto storeOp : storeOps) {
-    OpResult result = storeOp.getValue().cast<OpResult>();
-    SmallVector<OpFoldResult> resultOffsets, resultSizes;
-    if (failed(op.getResultTilePosition(rewriter, result.getResultNumber(),
-                                        offsets, sizes, resultOffsets,
-                                        resultSizes))) {
-      return rewriter.notifyMatchFailure(
-          op, "failed to rewrite destructive update");
+
+  std::vector<Operation *> sortedOps = sortOpsTopologically(producers);
+  return sortedOps;
+}
+
+/// Return all slices that are used to access a tile of the producer. Assume
+/// that `tiledOps` are in "reverse" order of their appearance in the IR.
+static SmallVector<tensor::ExtractSliceOp> getAllFusableProducerUses(
+    Operation *untiledOp, ArrayRef<Operation *> tiledOps) {
+  SmallVector<tensor::ExtractSliceOp> sliceOps;
+  for (auto tiledOp : llvm::reverse(tiledOps)) {
+    for (OpOperand &operand : llvm::reverse(tiledOp->getOpOperands())) {
+      auto sliceOp = operand.get().getDefiningOp<tensor::ExtractSliceOp>();
+      if (!sliceOp || sliceOp.getSource().getDefiningOp() != untiledOp)
+        continue;
+      sliceOps.push_back(sliceOp);
     }
-    SmallVector<OpFoldResult> resultStrides(resultOffsets.size(),
-                                            rewriter.getIndexAttr(1));
+  }
+  return sliceOps;
+}
 
-    SmallVector<OpFoldResult> combinedOffsets, combinedSizes, combinedStrides;
-    if (failed(IREE::Flow::foldOffsetsSizesAndStrides(
-            rewriter, loc, storeOp.getMixedOffsets(), storeOp.getMixedSizes(),
-            storeOp.getMixedStrides(), storeOp.getDroppedDims(), resultOffsets,
-            resultSizes, resultStrides, combinedOffsets, combinedSizes,
-            combinedStrides))) {
+FailureOr<TileAndFuseResult>
+TileAndFuseDispatchUsingSCFForOp::returningMatchAndRewrite(
+    TilingInterface op, PatternRewriter &rewriter) const {
+  TileAndFuseResult tileAndFuseResult;
+  // Apply the tiling pattern.
+  FailureOr<TilingResult> tilingResult =
+      tilingPattern.returningMatchAndRewrite(op, rewriter);
+  if (failed(tilingResult)) {
+    return failure();
+  }
+  // If there is no tiling then there is nothing to do for fusion.
+  if (!tilingResult->tiledLoops.any()) {
+    return tileAndFuseResult;
+  }
+  tileAndFuseResult.tiledAndFusedOps.push_back(tilingResult->tiledOp);
+  tileAndFuseResult.loops.append(tilingResult->loops);
+
+  // Get all ops to tile and fuse.
+  std::vector<Operation *> fusableProducers = getAllFusableProducers(op);
```

Comment:
Since the `op` has been tiled by `tilingPattern.returningMatchAndRewrite(op, rewriter)`, I guess it has been erased? (So the ASAN crashed)